packages/controller/src/routes.ts

```diff
@@ -39,7 +39,7 @@ function mergeSamples(destinationResult: lib.CollectorResultSerialized, sourceRe
 	}
 
 	for (let entry of receivedSamples) {
-		sourceResult.samples.push(entry);
+		destinationResult.samples.push(entry);
```

Comment:
This appears to be a bug fix, a regression test would be beneficial here.

Reply:
The commit changing the line describes it in detail. I don't think a regression test is worth the effort here because this line of code will likely never be hit.